receiver/hostmetricsreceiver/collector/cpu/cpu_collector_windows.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build windows
+
+package cpu
+
+import (
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal"
+)
+
+func (c *Collector) getPerCoreMetrics(reportPerCPU bool) []*metricspb.Metric {
+	return []*metricspb.Metric{
+		{
+			MetricDescriptor: MetricCPUSeconds,
+			Timeseries: []*metricspb.TimeSeries{
+				internal.GetDoubleTimeSeries(c.startTime, 0, LabelValueCPUUser, &metricspb.LabelValue{HasValue: false}),
+				internal.GetDoubleTimeSeries(c.startTime, 0, LabelValueCPUSystem, &metricspb.LabelValue{HasValue: false}),
+				internal.GetDoubleTimeSeries(c.startTime, 0, LabelValueCPUIdle, &metricspb.LabelValue{HasValue: false}),
+				internal.GetDoubleTimeSeries(c.startTime, 0, LabelValueCPUNice, &metricspb.LabelValue{HasValue: false}),
+				internal.GetDoubleTimeSeries(c.startTime, 0, LabelValueCPUIOWait, &metricspb.LabelValue{HasValue: false}),
+			},
+		},
+		{
+			MetricDescriptor: MetricCPUUtilization,
+			Timeseries:       []*metricspb.TimeSeries{internal.GetDoubleTimeSeries(c.startTime, 0, &metricspb.LabelValue{HasValue: false})},
+		},
+	}
```

Comment:
@rghetia 

I was reading some of the (unfinished) metric spec documentation which refers to generating metric instruments via the Meter interface:
https://github.com/jmacd/opentelemetry-specification/blob/2a74584326655699d7c607dc885be0aba348cd20/specification/api-metrics.md#meter-interface

I see most of this is already implemented in the Go SDK (although the proto itself refers to Gauges and not counters so maybe that's not up-to-date yet?):
https://github.com/open-telemetry/opentelemetry-go/tree/master/api

I also see we have the idea of Observer metric type which should presumably be used for cpu utilization rather than Gauge metric (which is going to be deprecated).

So, basically I'm a bit confused as to if this is correct - is it okay to just generate metrics directly like this for now, or should I be making an attempt to use what we have in the Go SDK instead?

Reply:
Generating metrics directly like this is fine. If SDK is used then you would need to convert from SDK format to internal format. So from that perspective this is better.